ecdsa: fix `-missing-prototypes` warning when `MBEDTLS_ECDSA_SIGN_ALT` is defined (#7415)
Mbed-TLS/mbedtls: ChangeLog.d/fix-declaration-of-mbedtls_ecdsa_sign_det_restartable-function.txt

Bugfix
   * Fix declaration of mbedtls_ecdsa_sign_det_restartable() function

>      in the ecdsa.h header file. The function was declared only when 
>      the macro MBEDTLS_ECDSA_SIGN_ALT is not defined and thus
>      producing a build warning when the macro MBEDTLS_ECDSA_SIGN_ALT

Reviewer: I think that's enough. 
```suggestion
     in the ecdsa.h header file. There is a build warning when the macro 
     MBEDTLS_ECDSA_SIGN_ALT is defined.
```
Please fix me if my suggestion looks strange.

Author: The suggestion looks good and have applied it.
